### PR TITLE
Various header additions for systemd

### PR DESCRIPTION
**Pull request by Dennisbonke**
HARD BLOCKED ON #766, #773, #776, #823, #824, #825, #826 (all have been merged!)

~~This PR is my internal staging area for systemd. I will cherry-pick commits into smaller chunks, that are actually reviewable, as this is of course not doable.~~

This was the systemd master staging PR, which has been split out in the PRs linked above. I found it fitting to finish with it. This PR adds some defines, adds one struct, and stubs 2 functions that don't need to be exported in a header.

Part of the systemd on Managarm project.

### CODE REVIEW

**Comment by Dennisbonke** [comment]
The last big part of the monster systemd PR is finally here! This PR finishes most of the legwork of getting `systemd` to compile. A few controversial changes have been left out, and some changes have been dropped in favor of work from the mlibc LFS project. This is part 8 of the systemd master PR, finalizing it.

Part of the systemd on Managarm project.

File: options/linux/generic/capabilities.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int capset(void *, void *) {
```
Comment (no92) [inline options/linux/generic/capabilities.cpp:5]: These functions don't seem to be part of libc, at least on my host. They aren't declared anywhere, either.

File: options/posix/include/net/if.h [hunk 2]
```diff
@@ -47,6 +48,22 @@ struct ifreq {
 		char *ifru_data;
 	} ifr_ifru;
 };
+#define ifr_name ifr_ifrn.ifrn_name
```
Comment (no92) [inline options/posix/include/net/if.h:51]: These defines are duplicate, see right below.

**Review by no92 — COMMENTED** [review]
Other than the two complaints, LGTM.

File: options/linux/generic/capabilities.cpp (hunk 1 above)
Comment (ikbenlike) [reply options/linux/generic/capabilities.cpp:5]: they're part of the linux capabilities API, should be declared in `sys/capability.h`

Comment (no92) [reply options/linux/generic/capabilities.cpp:5]: The headers seems to be part of `libcap` though.

Comment (ikbenlike) [reply options/linux/generic/capabilities.cpp:5]: they would have to be implemented as syscalls I think - implementing them here and providing an empty libcap object seems like the nicest option to me

Comment (Dennisbonke) [reply options/linux/generic/capabilities.cpp:5]: I followed musl here, which does this and then calls the linux syscall.

Comment (Dennisbonke) [reply options/linux/generic/capabilities.cpp:5]: See [this](https://github.com/bminor/musl/blob/7d756e1c04de6eb3f2b3d3e1141a218bb329fcfb/src/linux/cap.c#L8).

Comment (Geertiebear) [reply options/linux/generic/capabilities.cpp:5]: If these are stubbed then they should at least log a message that they are so.

Comment (Dennisbonke) [reply options/linux/generic/capabilities.cpp:7]: As requested, I've added a message.

File: options/posix/include/net/if.h (hunk 2 above)
Comment (Dennisbonke) [reply options/posix/include/net/if.h]: Dropped the offending commit.

**Review by Geertiebear — APPROVED** [review]
LGTM

**Pull request merged**